compiler/rustc_typeck/src/check/fn_ctxt/checks.rs

```diff
@@ -181,7 +186,25 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             } else {
                 false
             };
-            error = Some((expected_arg_count, supplied_arg_count, "E0061", sugg_unit));
+
+            // are we passing elements of a tuple without the tuple parentheses?
+            let expected_input_tys = if expected_input_tys.is_empty() {
```

Comment:
This is essentially duplicated on line 217.

Reply:
I guess this is because we need it for the call to `suggested_tuple_wrap`?

Reply:
I think this whole (pre-existing) function could be cleaned up a lot in general. Like, why does this `if` (and the one on L217) even need to exist? And why is an empty list used as a sentinel value?

Reply:
I'm thinking of trying to refactor it myself.

Reply:
Note that this gets changed quite a bit in #92364

Changing empty list as a sentinel is something orthogonal that I've thought about and worth doing.

Reply:
Yeah, the empty list led to a lot of confusion during the development of this too - like you say, it being used as a sentinel could be made more explicit (or even removed).